### PR TITLE
Run docker build (composer etc) as non root

### PR DESCRIPTION
**Pull request by leonk**
### Context

> Does this issue have a Trello card?

https://trello.com/c/9XtYmHgl/1442-update-php-version-to-8-as-74-is-eol-in-november-2022

### Intent

As prep for the above card, update the Dockerfile to run composer as a different user from root.
As well as being good security practise, this should also speed up the builds - as previously we were having to run `chown` on the entire directory after running composer, which would take quite a while.

### Considerations

> Is there any additional information that would help when reviewing this PR?

> Are there any steps required when merging/deploying this PR?

### Checklist

- [ ] This PR contains **only** changes related to the above card
- [ ] Tests have been added/updated to cover the change
- [ ] Documentation has been updated where appropriate
- [ ] Tested in Development

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -60,7 +60,7 @@ jobs:
     steps:
       - checkout
       - setup_remote_docker:
-          version: 19.03.13
+          version: 20.10.17
```
Comment (leonk) [inline .circleci/config.yml:63]: Updating to latest version of docker in circleci

File: Dockerfile [hunk 2]
```diff
@@ -129,27 +130,26 @@ COPY phpunit.xml phpunit.xml
...
-# Install dependencies (with dev)
+# Set to www-data user.
+USER 33
+
+RUN mkdir -p ~/phpunit/browser_output
+
+## Install dependencies (with dev)
 RUN composer install \
   --no-ansi \
-  --dev \
```
Comment (leonk) [inline Dockerfile:135]: --dev is now the default so removing it here.

File: composer.json [hunk 3]
```diff
@@ -78,7 +78,17 @@
...
+        "sort-packages": true,
+        "allow-plugins": {
+            "composer/installers": true,
+            "drupal/core-composer-scaffold": true,
+            "drupal/core-project-message": true,
+            "cweagans/composer-patches": true,
+            "drupal/console-extend-plugin": true,
+            "drupal/core-vendor-hardening": true,
+            "oomphinc/composer-installers-extender": true,
+            "dealerdirect/phpcodesniffer-composer-installer": true
```
Comment (leonk) [inline composer.json:90]: `allow-plugins` is now required in the latest version of composer.

**Review by BarryGeeMOJ — APPROVED** [review]
lgtm

**Review by BarryGeeMOJ — APPROVED** [review]
lgtm

**Pull request merged**